src/common/interfaces/IGasService.sol

```diff
@@ -1,6 +1,11 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity >=0.5.0;
 
+struct MessageCost {
+    uint32 messageSize;
+    uint128 gasCost;
+}
```

Comment:
FYI, I can not know the message size from the code, because some messages has dynamic size based on its content. This is why I need to add the size per action.